### PR TITLE
Split Track.cpp and Track.h

### PR DESCRIPTION
**Pull request by M374LX**
This PR splits Track.cpp and Track.h into one pair of files per class, as suggested in #5592.

### CODE REVIEW

**Comment by JohannesLorenz** [comment]
@M374LX This is just moving code between files, nothing else?

**Comment by M374LX** [comment]
@JohannesLorenz Other than moving code between files, there are a few changes in includes and forward declarations.

**Comment by JohannesLorenz** [comment]
Are you able to rebase this branch, so the mentioned conflicts will vanish?

**Comment by JohannesLorenz** [comment]
Let me know when I should review it.

Also, as you develop more PRs, just as information: We are currently trying to stop writing new PRs, in order to get our already started PRs closed for the reorganizing (#5592).

**Comment by M374LX** [comment]
@JohannesLorenz It is ready for review.

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:28]: Can you please replace this by including `QtGlobal` and change `assert` below to `Q_ASSERT`?

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:31]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:32]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:37]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:38]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:41]: All until (and including) BBTrackContainer.h are not used.

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:48]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:49]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:50]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:52]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:53]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:54]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:55]: not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:57]: All until (and including) SongEditor.h are not used

File: src/gui/TrackView.cpp
Comment (JohannesLorenz) [inline src/gui/TrackView.cpp:62]: Not used

File: src/gui/widgets/FxLineLcdSpinBox.cpp
Comment (JohannesLorenz) [inline src/gui/widgets/FxLineLcdSpinBox.cpp:30]: can be removed

**Review by JohannesLorenz — CHANGES_REQUESTED** [review]
Very useful PR, thanks for your efforts.

I left a lot of comments, but it's almost all about unused includes. Removing them will partly be a bit of work, but it will increase compile time.

File: include/BBEditor.h
Comment (M374LX) [reply include/BBEditor.h:32]: That is right. I should have added it to TrackContentWidget.cpp instead.

File: src/gui/TrackView.cpp
Comment (M374LX) [reply src/gui/TrackView.cpp:57]: Mixer is actually needed, as seen in ```TrackView::dropEvent()```.

Comment (JohannesLorenz) [reply src/gui/TrackView.cpp:57]: Good point :+1:

File: include/TrackContainerView.h
Comment (M374LX) [reply include/TrackContainerView.h:36]: This one gave me a lot of work, but I did it. :+1:

File: src/core/Track.cpp
Comment (JohannesLorenz) [inline src/core/Track.cpp:37]: I did not see it last review - Can you please fix the top comment (lines 2 and 3) and the doxygen `\file` comment (line 27)?

**Review by JohannesLorenz — APPROVED** [review]
Only one review comment added about some code comments. Functional it's all OK.

**Review by JohannesLorenz — APPROVED** [review]
LGTM!

**Comment by JohannesLorenz** [comment]
Thanks for the PR!

I'll wait for CI to finish and then I will "Squash and Merge". If you think it's too early for a merge because you want to add more commits for any reason, scream now :smiley:

**Comment by M374LX** [comment]
@JohannesLorenz I believe it is ready to be merged.

**Pull request merged**
